README.md
wgpc
====

West Granville Presbyterian Church


calendar/ChangeLog
Version 0.9.40 (30 Nov 2002)
  - Added support for all-day events that will prevent any events from
    being added to that day (good for out-of-the-office/vacation).
  - Added ability to limit the number of timed events (appointments)
    a single user can have on one day.  This feature can be turned
    on and the limit set in the admin system settings.
  - The ability to override an event conflict (from either a scheduling
    conflict or exceeding max number of events per day) is now
    configurable in the admin system settings.
  - Added support for external participants.  If an email address is
    provided with the external user name, they can receive notifications
    and reminders.  There are three new admin settings for this, one
    that enables external users, one that allows notifications for them,
    and one that allows email reminders for them.
  - Added system settings option that will allow new public events to
    not require any approval.  (Previously, this always required an
    admin user's approval.)
  - Fixed security bug where public access users could add "&user=craig"
    onto the URL to view calendars they weren't supposed to get to.
  - Added support for selection lists in site extras.  This can be used
    to provide a list of possible locations, etc.
  - Fixed bug: Despite setting "public user can view other users' calendars'
    to yes, still could not do this.
  - Added sql2html.pl tool for providing documentation about the database
    tables.  See README in the docs directory.
  - Added Welsh language translation
  - Added call to dbi_close() to pages that do printer-friendly versions
    (which do not include trailer.php, where to normal dbi_close() is.)
  - Added boss/assistant code that simplifies managing the calendar of
    another user.
  - Changed "ORDER BY UPPER(cal_name)" to "ORDER BY cal_name" in groups.php
    since older versions of MySQL do not like this.
  - Fixed bug 637318: Email all on participants on view_entry.php
    was not placing not including a space after the comma (which
    certain mail apps require.)
  - Fixed bug 616143: removed irrelevant line of code causing
    errors in week.php
  - Fixed bug 614273: unable to delete global categories
  - Added timezone offset support.
  - When editing system preferences for public access, use system settings
    for default values in pref.php.
  - Updated Portuguese_BR translation
  - Updated Dutch translation
  - Fixed bug 606332: '+' icons appeared in printer-friendly version of week
    view and day view.
  - Updated Spanish translation
  - Fixed bug 491366: If event goes past midnight, don't display 25:00.
  - Fixed bug 601006: Missing quote in URL notification for rejected event.
  - Fixed bug 597815: After setting preferences for the public user, each
    page had a SQL error at the bottom due to an incorrect entry being added
    to the webcal_user_pref table.
  - Added ability of an admin user to configure layers for public access user.
  - Fixed bug 595976, 583269: error generated in SQL when layers enabled
  - Updated FAQ.html with simpler instructions for configuring public access
    and http-based authentication.
  - Fixed bug: in day.php, events displayed at hour_arr[N] instead of
    the actual event name.
  - Updated Dutch translation
  - Fixed security bug: Don't allow users to put "login=XXX" in the URL
    to read someone else's calendar.
  - Use is_array test on categories to avoid error messsage.
  - Fixed bug: removed call by reference in php-dbi.php.
  - Fixed bug: Event description popup were not displayed in Netscape 4 if
    the event appeared near the bottom of the page.
  - Fixed bug 599323: categories were not being saved properly.
  - Fixed bug 601013: in week.php, leftarrow.gif had syntax error for border
    setting.
  - Added ability to set default for sending reminders to be yes when adding
    a new event.
  - Fixed bug 598947: background colors for today in week view were wrong.
  - Fixed bug 586436: duplicate table in tables-postgres.sql.
  - Updated German translation
  - Updated tables-db2.sql (was out-of-date).
  - Added + icon for adding new events on each hour for week and day views.
  - Fixed bug 598765: Fixed default date setting.
Version 0.9.39 (21 Aug 2002)
  - Fixed bug 597815: Fixed autoset of 'server_url' config setting to
    work when admin.php isn't the first page accessed.
  - Fixed bug 586525: Don't allow users to access views when view other
    users is set to no in admin.
  - Fixed bug 596616: date_to_string was corrupting certain month names
    in other languages.
  - Fixed bug 587795: Could not edit a layer with less than two users
    in the calendar.
  - Changed login page so that "Remember login" is not selected by default.
  - Fixed bug: Was not defaulting login name to last person to login.
  - Fixed bug: Events with no time on weekends where not being displayed
    with the weekend background color in the week view.
  - Include a category of "None" when editing an event.
  - Fixed bug 587909: Don't display 12pm for events with no time in
    notifications.
  - Fixed bug 585939: Don't allow empty view names or group names.
  - Fixed bug: Some users encounter an error when logging in because
    the old session cookie (that contained no path info) was not
    being deleted properly.
  - Fixed bug: send_reminders.php was not checking single user status
    properly.
  - Fixed bug: Error in mailto URL in view_entry.php.
  - Fixed bug: Variable $i was being used for both inner and outer for
    loop in del_entry.php.
  - Fixed bug: When adding/editing an event, if you put a leading '0'
    on the start time (like 09), it always warns you that the event
    starts before your working hours (even if it isn't true).

calendar/login.php
<?php
include "includes/config.php";
include "includes/php-dbi.php";
include "includes/functions.php";
include "includes/$user_inc";
include "includes/connect.php";

// Change this to true to show "no such user" or "invalid password" on
// login failures.
$showLoginFailureReason = true;

load_global_settings ();

if ( ! empty ( $last_login ) ) {
  $login = "";
}

if ( empty ( $webcalendar_login ) ) {
  $webcalendar_login = "";
}

if ( $remember_last_login == "Y" && empty ( $login ) ) {
  $last_login = $login = $webcalendar_login;
}


include "includes/translate.php";

// see if a return path was set
if ( ! empty ( $return_path ) ) {
  $return_path = clean_whitespace ( $return_path );
  $url = $return_path;
} else {
  $url = "index.php";
}

$lang = '';
if ( ! empty ( $LANGUAGE ) &&  $LANGUAGE != "Browser-defined" && $LANGUAGE != "none" ) {
  $lang = languageToAbbrev ( $LANGUAGE );
} else {
  $lang_long = get_browser_language ();
  $lang = languageToAbbrev ( $lang_long );
}

if ( empty ( $lang ) ) {
  $lang = 'en';
}

$login = getPostValue ( 'login' );
$password = getPostValue ( 'password' );

if ( ! empty ( $settings['session'] ) && $settings['session'] = 'php' ) {
  session_start ();
}

// calculate path for cookie
if ( empty ( $PHP_SELF ) ) {
  $PHP_SELF = $_SERVER["PHP_SELF"];
}
$cookie_path = str_replace ( "login.php", "", $PHP_SELF );
//echo "Cookie path: $cookie_path\n$cookie_path1";

if ( $single_user == "Y" ) {
  // No login for single-user mode
  do_redirect ( "index.php" );
} else if ( $use_http_auth ) {
  // There is no login page when using HTTP authorization
  do_redirect ( "index.php" );
} else {
  if ( ! empty ( $login ) && ! empty ( $password ) ) {
    if ( get_magic_quotes_gpc() ) {
      $password = stripslashes ( $password );
      $login = stripslashes ( $login );
    }
    $login = trim ( $login );
    if ( $login != addslashes ( $login ) ) {
      die_miserable_death ( "Illegal characters in login " .
        "<tt>" . htmlentities ( $login ) . "</tt>" );
    }
    if ( user_valid_login ( $login, $password ) ) {
      user_load_variables ( $login, "" );
      // set login to expire in 365 days
      srand((double) microtime() * 1000000);
      $salt = chr( rand(ord('A'), ord('z'))) . chr( rand(ord('A'), ord('z')));
      $encoded_login = encode_string ( $login . "|" . crypt($password, $salt) );

      if ( ! empty ( $settings['session'] ) && $settings['session'] = 'php' ) {
        $_SESSION['webcalendar_session'] = $encoded_login;
      } else {
        if ( ! empty ( $remember ) && $remember == "yes" ) {
          SetCookie ( "webcalendar_session", $encoded_login,
            time() + ( 24 * 3600 * 365 ), $cookie_path );
        } else {
          SetCookie ( "webcalendar_session", $encoded_login, 0, $cookie_path );
        }
      }
load_user_preferences ();
      // The cookie "webcalendar_login" is provided as a convenience to
      // other apps that may wish to find out what the last calendar
      // login was, so they can use week_ssi.php as a server-side include.
      // As such, it's not a security risk to have it un-encoded since it
      // is not used to allow logins within this app.  It is used to
      // load user preferences on the login page (before anyone has
      // logged in) if $remember_last_login is set to "Y" (in admin.php).
      if ( ! empty ( $remember ) && $remember == "yes" ) {
        SetCookie ( "webcalendar_login", $login,
          time() + ( 24 * 3600 * 365 ), $cookie_path );
      } else {
        SetCookie ( "webcalendar_login", $login, 0, $cookie_path );
      }
      do_redirect ( $url );
    } else {
      // Invalid login
      if ( empty ( $error ) || ! $showLoginFailureReason ) {
        $error = translate("Invalid login" );
      }
    }
  } else {
    // No login info... just present empty login page
    //$error = "Start";
  }
  // delete current user
  SetCookie ( "webcalendar_session", "", 0, $cookie_path );
  // In older versions the cookie path had no trailing slash and NS 4.78
  // thinks "path/" and "path" are different, so the line above does not
  // delete the "old" cookie. This prohibits the login. So we delete the
  // cookie with the trailing slash removed
  if (substr($cookie_path, -1) == '/') {
    SetCookie ( "webcalendar_session", "", 0, substr($cookie_path, 0, -1)  );
  }
}
$charset = ( ! empty ( $LANGUAGE )?translate("charset"): "iso-8859-1" );
echo "<?xml version=\"1.0\" encoding=\"$charset\"?>" . "\n";
?>
<!DOCTYPE html
    PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="<?php echo $lang; ?>" lang="<?php echo $lang; ?>">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=<?php echo $charset; ?>" />
<title><?php etranslate($application_name)?></title>
<script type="text/javascript">
// error check login/password
function valid_form ( form ) {
  if ( form.login.value.length == 0 || form.password.value.length == 0 ) {
    alert ( "<?php etranslate("You must enter a login and password")?>." );
    return false;
  }
  return true;
}
function myOnLoad() {
  <?php if ( ! empty ( $plugins_enabled ) && ( $plugins_enabled ) ){ ?>
   if (self != top)  {
     window.open("login.php","_top","");
     return;
   }
  <?php } ?>
  document.login_form.login.focus();
  <?php
    if ( ! empty ( $login ) ) echo "document.login_form.login.select();";
    if ( ! empty ( $error ) ) {
      echo "  alert ( \"$error\" );\n";
    }
  ?>
}
</script>
<?php 
 include "includes/styles.php";

 // Print custom header (since we do not call print_header function)
 if ( ! empty ( $CUSTOM_SCRIPT ) && $CUSTOM_SCRIPT == 'Y' ) {
   $res = dbi_query (
     "SELECT cal_template_text FROM webcal_report_template " .
     "WHERE cal_template_type = 'S' and cal_report_id = 0" );
   if ( $res ) {
     if ( $row = dbi_fetch_row ( $res ) ) {
       echo $row[0];
     }
     dbi_free_result ( $res );
   }
 }
?>
</head>
<body onload="myOnLoad();">
<?php
// Print custom header (since we do not call print_header function)
if ( ! empty ( $CUSTOM_HEADER ) && $CUSTOM_HEADER == 'Y' ) {
  $res = dbi_query (
    "SELECT cal_template_text FROM webcal_report_template " .
    "WHERE cal_template_type = 'H' and cal_report_id = 0" );
  if ( $res ) {
    if ( $row = dbi_fetch_row ( $res ) ) {
      echo $row[0];
    }
    dbi_free_result ( $res );
  }
}
?>

<h2><?php 
// If Application Name is set to Title then get translation
// If not, use the Admin defined Application Name
if ( ! empty ( $application_name ) &&  $application_name =="Title") {
  etranslate($application_name);
} else {
  echo htmlspecialchars ( $application_name );
}  
?></h2>

<?php
if ( ! empty ( $error ) ) {
  print "<span style=\"color:#FF0000; font-weight:bold;\">" . 
    translate("Error") . ": $error</span><br />\n";
} else {
  print "<br />\n";
}
?>
<form name="login_form" id="login" action="login.php" method="post" 
  onsubmit="return valid_form(this)">
<?php
if ( ! empty ( $return_path ) ) {
  echo "<input type=\"hidden\" name=\"return_path\" value=\"" .
    htmlentities ( $return_path ) . "\" />\n";
}
?>

<table cellpadding="10" align="center">
<tr><td rowspan="2">
 <img src="login.gif" alt="Login" /></td><td align="right">
 <label for="user"><?php etranslate("Username")?>:</label></td><td>
 <input name="login" id="user" size="15" maxlength="25" 
   value="<?php if ( ! empty ( $last_login ) ) echo $last_login;?>" 
   tabindex="1" />
</td></tr>
<tr><td style="text-align:right;">
 <label for="password"><?php etranslate("Password")?>:</label></td><td>
 <input name="password" id="password" type="password" size="15" 
   maxlength="30" tabindex="2" />
</td></tr>
<tr><td colspan="3" style="font-size: 10px;">
 <input type="checkbox" name="remember" id="remember" tabindex="3" 
   value="yes" <?php if ( ! empty ( $remember ) && $remember == "yes" ) {
     echo "checked=\"checked\""; }?> /><label for="remember">&nbsp;
   <?php etranslate("Save login via cookies so I don't have to login next time")?></label>
</td></tr>
<tr><td colspan="4" style="text-align:center;">
 <input type="submit" value="<?php etranslate("Login")?>" tabindex="4" />
</td></tr>
</table>
</form>

<?php if ( ! empty ( $public_access ) && $public_access == "Y" ) { ?>
 <br /><br />
 <a class="nav" href="index.php">
   <?php etranslate("Access public calendar")?></a><br />
<?php } ?>

<?php if ( $demo_mode == "Y" ) {
 // This is used on the sourceforge demo page
 echo "Demo login: user = \"demo\", password = \"demo\"<br />";
} ?>
<br /><br /><br />
<span class="cookies"><?php etranslate("cookies-note")?></span><br />
<hr />
<br /><br />
<?php $PROGRAM_NAME = "WebCalendar v1.0.5 (04 Mar 2007)"; ?>
<a href="<?php echo $PROGRAM_URL ?>" id="programname"><?php echo $PROGRAM_NAME?></a>

<?php // Print custom trailer (since we do not call print_trailer function)
if ( ! empty ( $CUSTOM_TRAILER ) && $CUSTOM_TRAILER == 'Y' ) {
  $res = dbi_query (
    "SELECT cal_template_text FROM webcal_report_template " .
    "WHERE cal_template_type = 'T' and cal_report_id = 0" );
  if ( $res ) {
    if ( $row = dbi_fetch_row ( $res ) ) {
      echo $row[0];
    }
    dbi_free_result ( $res );
  }
} ?>
</body>
</html>
